src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -1168,11 +1189,46 @@ private Pipe CreateRequestBodyPipe(uint windowSize)
         }
     }
 
+    public override async ValueTask<IWebTransportSession> AcceptAsync(CancellationToken token)
+    {
+        if (_isWebTransportSessionAccepted)
+        {
+            throw new InvalidOperationException(CoreStrings.AcceptCannotBeCalledMultipleTimes);
+        }
+        _isWebTransportSessionAccepted = true;
```

Comment:
Move this after EnableWebTransportAndH3Datagrams and IsWebTransportRequest. No state should be changed before those exceptions could be thrown.